### PR TITLE
Add KeyVault support to Scale Set deployment

### PR DESCRIPTION
**Pull request by scotmoor**
Access user name and password for VMs from KeyVault when deploying Azure Scale Set

### CODE REVIEW

**Comment by scotmoor** [comment]
@spinnaker/netflix-reviewers PTAL

**Comment by scotmoor** [comment]
@spinnaker/netflix-reviewers ping

**Comment by scotmoor** [comment]
@ajordens @duftler - PTAL

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy [hunk 1]
```diff
@@ -255,6 +252,19 @@ class AzureResourceManagerClient extends AzureBaseClient {
       body
   }
 
+  static String convertParametersToTemplateJSON(ObjectMapper mapper, Map<String, Object> sourceParameters) {
+    def parameters = [:]
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy]: Could use `parameters = sourceParameters.collectEntries { ... }` here.

Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy]: Feel free to make this change in a later PR.

**Pull request merged**